### PR TITLE
7 Ways to Deal with Application Secrets in Azure

### PR DESCRIPTION
**Pull request by mikhailshilkov**
The article was already reviewed but there's no publish date yet, so I'm opening the PR for a future date

### CODE REVIEW

File: content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
+---
+title: "7 Ways to Deal with Application Secrets in Azure"
+authors: ["mikhail-shilkov"]
+tags: ["Azure", "Security"]
+date: "2019-07-26"
+
+meta_image: "feature.jpg"
+---
+
+Every non-trivial application relies on configuration values that may depend on the current execution environment. Some of these values contain sensitive information which shouldn't be shared publicly. The fewer parties have access to those secret values, the safer the application is. Ideally, no one should be granted access at all.
```
Comment (ericrudder) [inline content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md]: Every non-trivial application relies on configuration values that may depend on the current execution environment. Some of these values contain sensitive information that shouldn't be shared publicly. In general, the fewer parties that have access to those secret values, the safer the application will be -- in fact, in an ideal world, no one would be granted direct access to those secrets.

File: content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+<!--more-->
+
+Examples of such secret configuration values include
```
Comment (ericrudder) [inline content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md]:
```suggestion
Examples of secret configuration values include:
```

File: content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+- A connection string to a message bus or a database;
```
Comment (ericrudder) [inline content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md]:
```suggestion
- A connection string to a message bus or a database
```

File: content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+- A SAS Token to an Azure Storage account;
```
Comment (ericrudder) [inline content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md]:
```suggestion
- A SAS Token to an Azure Storage account
```

File: content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+- An access key for a third-party service.
```
Comment (ericrudder) [inline content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md]:
```suggestion
- An access key for a third-party service
```

File: content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+
+There's no one universal way to manage the secrets, as a lot depends on the context. In this article, I go through seven ways to use secret values in a .NET Core application running in Azure. I start with naively hard-coded strings and build up from there to more secure options.
```
Comment (ericrudder) [inline content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md]:
```suggestion
There's no one universal way to manage secrets, as a lot depends on the context in which they are used. In this article, I go through seven ways to use secret values in a .NET Core application running in Azure. I start with naively hard-coded strings and build up from there to more secure options.
```

File: content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+
+``` cs
+var connectionString = "DefaultEndpointsProtocol=https;AccountName=account-name;AccountKey=account-key";
+var storageAccount = CloudStorageAccount.Parse(connectionString);
+var queueClient = storageAccount.CreateCloudQueueClient();
+```
+
+Both snippets are fine for a Hello-World application with a lifespan of two hours. However, I would strongly discourage doing so in any code that can potentially be checked into a source control system, even for a 10-minute experiment. Don't copy-paste secrets into files which are part of a git repository. One accidental `git commit & git push`&mdash;and the secrets are compromised.
+
+Avoid the "Secrets as Code" practice: there are bots scanning your GitHub for those.
```
Comment (ericrudder) [inline content/blog/7-ways-to-deal-with-application-secrets-in-azure/index.md]:
```suggestion
Avoid the "Secrets as Code" practice: there are bots scanning your GitHub repository for those.
```

**Comment by mikhailshilkov** [comment]
@ericrudder Thanks for the suggestions! I've updated the text.

**Pull request merged**